subsys/bluetooth/common/bt_to_str.h

```diff
@@ -0,0 +1,22 @@
+/* Copyright (c) 2022 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/uuid.h>
+
+/* NOTE: These helper functions always encodes into the same buffer storage.
+ * It is the responsibility of the user of this function to copy the information
+ * in this string if needed.
+ *
+ * NOTE: These functions are not thread-safe!
+ */
+const char *bt_hex_real(const void *buf, size_t len);
+const char *bt_addr_str_real(const bt_addr_t *addr);
+const char *bt_addr_le_str_real(const bt_addr_le_t *addr);
+const char *bt_uuid_str_real(const struct bt_uuid *uuid);
+
+#define bt_hex(buf, len) bt_hex_real(buf, len)
+#define bt_addr_str(addr) bt_addr_str_real(addr)
+#define bt_addr_le_str(addr) bt_addr_le_str_real(addr)
+#define bt_uuid_str(uuid) bt_uuid_str_real(uuid)
```

Comment:
I've always wondered: Is there a good reason why these are not available in the public API? 

Converting an addr to a string representation is something that we often do in samples an applications.

Reply:
Good point, I don't know either. We can consider doing that but in a separate PR.

Reply:
I think a good reason is that they are unsafe in concurrent code, and easy to misuse in general due to the static variable inside them. I have an idea for improved versions of these macros, which would allocate space on the stack before calling the real function. I think this version is safer and could be made public.

```c
/** Like bt_addr_le_to_str, but returns the pointer `out` instead of the write-length. */
char *bt_addr_le_str_ret(const bt_addr_le_t *addr, char *out)
{
	bt_addr_le_to_str(addr, out, BT_ADDR_LE_STR_LEN);
	return out;
}

/** The scope/lifetime of the returned string tied to the enclosing block.
  *
  * It can be safely used multiple times in a single statement and is thread-safe.
  *
  * `foo(bt_addr_le_str(addr))` equivalent with `char x[BT_ADDR_LE_STR_LEN];
  * foo(bt_addr_le_str_ret(addr, &x));`, but `x` is anonymous, so it cannot crash
  * with or shadow other definitions.
  */
#define bt_addr_le_str(addr) bt_addr_le_str_ret(addr, &(char[BT_ADDR_LE_STR_LEN]){})
```